omniscidb/ResultSet/ResultSet.cpp

```diff
@@ -1245,6 +1246,118 @@ void ResultSet::initializeStorage() const {
   }
 }
 
+void ResultSet::finalizeAggregates() {
+  auto timer = DEBUG_TIMER(__func__);
+  if (getQueryDescriptionType() == QueryDescriptionType::Projection) {
+    return;
+  }
+
+  // For columnar output holds slot indices, otherwise slot offsets.
+  std::vector<std::tuple<const TargetInfo*, size_t>> quantile_slots_to_finalize;
+  size_t cur_slot_idx = 0;
+  size_t cur_slot_offs = 0;
+  // Identify columns with aggregates requiring finalization.
+  for (size_t i = 0; i < targets_.size(); ++i) {
+    auto& tinfo = targets_[i];
+    // TODO: finalize TopK?
+    if (tinfo.is_agg && tinfo.agg_kind == hdk::ir::AggType::kQuantile) {
+      CHECK_EQ(query_mem_desc_.getPaddedSlotWidthBytes(cur_slot_idx), 8);
+      CHECK(lazy_fetch_info_.empty() || !lazy_fetch_info_[i].is_lazily_fetched);
+      if (query_mem_desc_.didOutputColumnar()) {
+        quantile_slots_to_finalize.emplace_back(std::make_tuple(&tinfo, cur_slot_idx));
+      } else {
+        quantile_slots_to_finalize.emplace_back(std::make_tuple(&tinfo, cur_slot_offs));
+      }
+    }
+    if (!query_mem_desc_.didOutputColumnar()) {
+      cur_slot_offs = advance_target_ptr_row_wise(cur_slot_offs,
+                                                  tinfo,
+                                                  cur_slot_idx,
+                                                  query_mem_desc_,
+                                                  separate_varlen_storage_valid_);
+    }
+    cur_slot_idx = advance_slot(cur_slot_idx, tinfo, separate_varlen_storage_valid_);
+  }
+
+  if (quantile_slots_to_finalize.empty()) {
+    return;
+  }
+
+  // This function is going to be called for each non-empty entry.
+  auto finalize = [&](const ResultSetStorage* storage, size_t entry_idx) {
+    for (auto& slot : quantile_slots_to_finalize) {
+      const int8_t* slot_ptr = nullptr;
+      if (query_mem_desc_.didOutputColumnar()) {
+        auto col_ptr =
+            storage->buff_ + storage->query_mem_desc_.getColOffInBytes(std::get<1>(slot));
+        slot_ptr = col_ptr + 8 * entry_idx;
```

Comment:
What does 8 stand for?

Reply:
The slot size that was checked earlier to be 8.